commit 8e01019c19682ceecc3ccd58e31752fa53f44316
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Fri Jul 3 15:42:25 2009 +0000

    Fix build of i965_drv_video.so (Paul Bender).

commit ce9e138bd91c630eca1bc24b91d1d2afbba8c781
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Wed Jun 24 10:59:06 2009 +0000

    Split libva DSO into core (libva.so) and display-dependent parts (libva-x11.so).

diff --git a/Makefile.am b/Makefile.am
index 57c07b3..189cba8 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -24,11 +24,16 @@ AUTOMAKE_OPTIONS = foreign
 SUBDIRS = dri src dummy_drv_video i965_drv_video test
 
 pcfiles = \
-	libva.pc
+	libva.pc \
+	libva-$(LIBVA_DISPLAY).pc
+
+# libva-<display>.pc - for display-specific dependencies
+libva-$(LIBVA_DISPLAY).pc: libva_display.pc
+	@cp $< $@
 
 pkgconfigdir = @pkgconfigdir@
 pkgconfig_DATA = $(pcfiles)
 
-EXTRA_DIST = libva.pc.in
+EXTRA_DIST = libva.pc.in libva_display.pc.in
 
 CLEANFILES = $(pcfiles)
diff --git a/configure.ac b/configure.ac
index 4532359..a154ac8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -90,8 +90,12 @@ AC_SUBST(LIBVA_LIBS)
 pkgconfigdir=${libdir}/pkgconfig
 AC_SUBST(pkgconfigdir)
 
-libvabackendlib=libva.la
+LIBVA_DISPLAY=x11
+libvacorelib=libva.la
+libvabackendlib=libva-$LIBVA_DISPLAY.la
+AC_SUBST([libvacorelib])
 AC_SUBST([libvabackendlib])
+AC_SUBST(LIBVA_DISPLAY)
 
 AC_OUTPUT([
 	Makefile
@@ -105,5 +109,6 @@ AC_OUTPUT([
 	i965_drv_video/shaders/render/Makefile
 	test/Makefile
 	libva.pc
+	libva_display.pc
 ])
 
diff --git a/libva_display.pc.in b/libva_display.pc.in
new file mode 100644
index 0000000..c9c5cd6
--- /dev/null
+++ b/libva_display.pc.in
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+display=@LIBVA_DISPLAY@
+
+Name: libva-${display}
+Description: Userspace Video Acceleration (VA) ${display} interface
+Version: @PACKAGE_VERSION@
+Libs: -L${libdir} -lva-${display}
+Cflags: -I${includedir}
diff --git a/src/Makefile.am b/src/Makefile.am
index 5a2d237..b763726 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -28,10 +28,18 @@ INCLUDES = \
 LDADD = \
 	$(LIBVA_LT_LDFLAGS)
 
-libva_la_LTLIBRARIES = libva.la
+lib_LTLIBRARIES = \
+	libva.la \
+	libva-x11.la
+
 libva_ladir = $(libdir)
 libva_la_LDFLAGS = $(LDADD) -no-undefined
-libva_la_LIBADD = $(LIBVA_LIBS) -ldl -lX11 -lXext x11/libva_x11.la -ldrm -lXfixes
+libva_la_LIBADD = $(LIBVA_LIBS) -ldl
+
+libva_x11_la_SOURCES = 
+libva_x11_la_LIBADD  = $(libvacorelib) x11/libva_x11.la $(LIBVA_LIBS) $(X11_LIBS) $(XEXT_LIBS) $(DRM_LIBS) $(XFIXES_LIBS)
+libva_x11_la_LDFLAGS = $(LDADD)
+libva_x11_la_DEPENDENCIES = $(libvacorelib)
 
 nodist_libva_la_SOURCES = va_version.h
 BUILT_SOURCES = va_version.h
diff --git a/src/x11/va_x11.c b/dri/va_x11.c
index f47c4d7..c184122 100644
--- a/src/x11/va_x11.c
+++ b/src/x11/va_x11.c
@@ -284,27 +284,6 @@ static VAStatus va_DisplayContextGetDriverName (
     return vaStatus;
 }
 
-int vaDisplayIsValid(VADisplay dpy)
-{
-  VADisplayContextP tmp=NULL;
-  VADisplayContextP pDisplayContext = pDisplayContexts;
-
-  while (pDisplayContext)
-  {
-      if (pDisplayContext == (VADisplayContextP)dpy)
-      {
-          tmp = (VADisplay)pDisplayContext;
-          break;
-      }
-      pDisplayContext = pDisplayContext->pNext;
-  }
-
-  if (!tmp)
-      return 0;
-  
-  return tmp->vaIsValid(pDisplayContext);
-}
-
 
 VADisplay vaGetDisplay (
     Display *native_dpy /* implementation specific */
diff --git a/src/va.c b/src/va.c
index 07d21a0..b956e85 100644
--- a/src/va.c
+++ b/src/va.c
@@ -52,7 +52,11 @@
 
 static int va_debug_trace = 0;
 
-int vaDisplayIsValid(VADisplay dpy);
+static int vaDisplayIsValid(VADisplay dpy)
+{
+  VADisplayContextP pDisplayContext = (VADisplayContextP)dpy;
+  return pDisplayContext && pDisplayContext->vaIsValid(pDisplayContext);
+}
 
 static void va_errorMessage(const char *msg, ...)
 {
diff --git a/i965_drv_video/Makefile.am b/i965_drv_video/Makefile.am
index b3d662c..a0bedca 100644
--- a/i965_drv_video/Makefile.am
+++ b/i965_drv_video/Makefile.am
@@ -27,7 +27,7 @@ AM_CFLAGS = -Wall -I$(top_srcdir)/src -I$(top_srcdir)/dri @DRM_CFLAGS@ -DIN_LIBV
 i965_drv_video_la_LTLIBRARIES = i965_drv_video.la
 i965_drv_video_ladir = @LIBVA_DRIVERS_PATH@
 i965_drv_video_la_LDFLAGS = -module -avoid-version -no-undefined -Wl,--no-undefined @DRM_LIBS@ -ldrm_intel
-i965_drv_video_la_LIBADD = ../src/libva.la -lpthread
+i965_drv_video_la_LIBADD = ../src/libva-x11.la -lpthread
 
 i965_drv_video_la_SOURCES =	\
 	object_heap.c		\
